rust/src/utils/kmeans.rs

```diff
@@ -172,6 +172,30 @@ struct KMeanMembership {
     metric_type: MetricType,
 }
 
+/// Split the largest clusters.
+///
+fn split_clusters(centroids: Vec<Float32Array>, histogram: Vec<i32>) -> Result<Vec<Float32Array>> {
+    // Used to split the centroid of largest cluster to two centroids.
+    // See Faiss
+    const EPS: f32 = 1.0 / 1024.0;
+
+    let mut histogram = histogram;
+    let mut centroids = centroids;
+    // Split the biggest clusters
+    for i in 0..histogram.len() {
+        if histogram[i] == 0 {
```

Comment:
curious if any of the clusters in our test data actually gets split

Reply:
For the dataset i was running on, it had 1 split on global IVF kmeans, about 10-20 splits at each local pq training.